src/adapter/src/coord/timeline.rs

```diff
@@ -190,34 +197,67 @@ impl Coordinator {
         timeline: &'a Timeline,
         initially: Timestamp,
         now: NowFn,
+        timestamp_oracle_impl: TimestampOracleImpl,
         timestamp_persistence: P,
+        timestamp_oracle_url: Option<String>,
+        metrics: &Arc<timestamp_oracle::metrics::Metrics>,
         global_timelines: &'a mut BTreeMap<Timeline, TimelineState<Timestamp>>,
     ) -> &'a mut TimelineState<Timestamp>
     where
         P: TimestampPersistence<mz_repr::Timestamp> + 'static,
     {
         if !global_timelines.contains_key(timeline) {
-            let oracle = if timeline == &Timeline::EpochMilliseconds {
-                CatalogTimestampOracle::new(
-                    initially,
-                    move || (now)().into(),
-                    *TIMESTAMP_PERSIST_INTERVAL,
-                    timestamp_persistence,
-                )
-                .await
+            info!(
+                "opening a new {:?} TimestampOracle for timeline {:?}",
+                timestamp_oracle_impl, timeline,
+            );
+
+            let now_fn = if timeline == &Timeline::EpochMilliseconds {
+                now
             } else {
-                CatalogTimestampOracle::new(
-                    initially,
-                    Timestamp::minimum,
-                    *TIMESTAMP_PERSIST_INTERVAL,
-                    timestamp_persistence,
-                )
-                .await
+                NOW_ZERO.clone()
```

Comment:
I realize this is effectively code movement, but seems like a nice time to add a comment on what's going on here

Reply:
will do, and I'll also change this to use `Timestamp::minimum`, like before, to make the intent clearer